mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -18,28 +18,46 @@
 package org.apache.spark.ml.linalg
 
 import com.github.fommil.netlib.{BLAS => NetlibBLAS, F2jBLAS}
-import com.github.fommil.netlib.BLAS.{getInstance => NativeBLAS}
+import scala.util.Try
+
+import org.apache.spark.util.Utils
 
 /**
  * BLAS routines for MLlib's vectors and matrices.
  */
 private[spark] object BLAS extends Serializable {
 
-  @transient private var _f2jBLAS: NetlibBLAS = _
+  @transient private var _javaBLAS: NetlibBLAS = _
   @transient private var _nativeBLAS: NetlibBLAS = _
   private val nativeL1Threshold: Int = 256
```

Comment:
Will we need to change this value depending on which javaBLAS is selected? vectorized version or F2JBLAS.

Reply:
From the results of the benchmarks (see https://github.com/apache/spark/pull/30810#issuecomment-747405702), the value is likely to be a bit bigger when using vectorized implementation and we will keep something similar for some operations (`dscal`, `dspr`, `dsyr`, `sdot`).

I'll look at why is the native implementation so much faster for simple operations like `dscal` and `sdot` (especially when `ddot` is equivalent to native), and what we can apply to the vectorized version.